modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -527,6 +531,463 @@ public void testAddCustomObjectField() throws Exception {
+						ObjectFieldConstants.READ_ONLY_CONDITIONAL
+					).build())));
+
+		String invalidDDMscript = RandomTestUtil.randomString() + "()";
+
+		AssertUtils.assertFailure(
+			ObjectFieldReadOnlyConditionExpressionException.class,
+			"Syntax error in: " + invalidDDMscript,
+			() -> ObjectDefinitionTestUtil.addObjectDefinition(
+				false, _objectDefinitionLocalService,
+				Arrays.asList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"a" + RandomTestUtil.randomString()
+					).readOnly(
+						ObjectFieldConstants.READ_ONLY_CONDITIONAL
+					).readOnlyConditionExpression(
+						invalidDDMscript
+					).build())));
+
+		ObjectDefinition objectDefinition1 =
+			ObjectDefinitionTestUtil.addObjectDefinition(
+				false, _objectDefinitionLocalService, Collections.emptyList());
+
+		ObjectField objectField = _objectFieldLocalService.getObjectField(
+			objectDefinition1.getObjectDefinitionId(), "creator");
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _objectFieldLocalService.getObjectField(
+			objectDefinition1.getObjectDefinitionId(), "externalReferenceCode");
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_FALSE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).readOnly(
+				null
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_FALSE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).readOnly(
+				ObjectFieldConstants.READ_ONLY_TRUE
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).readOnly(
+				ObjectFieldConstants.READ_ONLY_CONDITIONAL
+			).readOnlyConditionExpression(
+				"isEmpty(able)"
```

Comment:
You can create a variable to store this value, you are using it eight times.
Do you agree?